services/resource_dereference_service.go

```diff
@@ -0,0 +1,101 @@
+package services
+
+import (
+	// jsonpb Marshaller is deprecated, but is needed because there's only one way to proto
+	// marshal in combination with our proto generator version
+	"encoding/json"
+
+	"github.com/cheqd/did-resolver/types"
+	"github.com/cheqd/did-resolver/utils"
+)
+
+type ResourceDereferenceService struct {
+	ledgerService LedgerServiceI
+	didDocService DIDDocService
+}
+
+func NewResourceDereferenceService(ledgerService LedgerServiceI, didDocService DIDDocService) ResourceDereferenceService {
+	return ResourceDereferenceService{
+		ledgerService: ledgerService,
+		didDocService: didDocService,
+	}
+}
+
+func (rds ResourceDereferenceService) DereferenceResource(path string, did string, didUrl string, dereferenceOptions types.DereferencingOption) (types.DidDereferencing, int) {
```

Comment:
Why do we need path and did here? Can't we extract them from didUrl?

Reply:
Done